stacker/actions/build.py

```diff
@@ -267,11 +267,14 @@ def _generate_plan(self, tail=False):
         stacks = self.context.get_stacks_dict()
         dependencies = self._get_dependencies()
         for stack_name in self.get_stack_execution_order(dependencies):
-            plan.add(
-                stacks[stack_name],
-                run_func=self._launch_stack,
-                requires=dependencies.get(stack_name),
-            )
+            try:
```

Comment:
It would probably be safer to only wrap the call that can throw the Exception you're seeing - in this case, I believe it's the call to `stacks[stack_name]` not something inside `plan.add`.  If that's the case, I'd change this to:

```
try:
    stack = stacks[stack_name]
except KeyError:
    raise StackDoesNotExist(stack_name)

plan.add(
    stack,
    ...
)
```

What do you think?

Reply:
@phobologic, yeah.. You are correct in the issue is stacks[stack_name] and a more simply way to check this would be better than wrapping the whole plan.add.